### DISCUSSION TITLE
What is the workflow for generating an assembled executable?

### DISCUSSION THREAD

**Discussion by NomisIV** [icedland/iced#256] [Q&A] [answered]
I'm very new to assemby programming and I'm trying to generate a simple hello world program using this project (with rust). I'm using the `CodeAssembler` to generate assembly and then encode it with the `assemble` method to encode the instructions to bytes. Then I assumed that these bytes would be the final executable, so I tried writing them to a file, making it executable and then running it, but I got the following error from my shell: `zsh: exec format error: ./out`. After that I tried linking it with `ld`, but `ld` didn't recognize the format. So now I'm out of ideas. How do I generate an executable with this crate?

**Comment by wtfsck** [comment, marked as answer]
Sounds like you want to use a tool that generates a binary (eg. PE or ELF) with code. This lib doesn't do that. Try [nasm](https://github.com/netwide-assembler/nasm) instead.

**Reply by NomisIV** [reply]
Ah, alright, thanks!
